src/navigation/AddWalletNavigator.tsx

```diff
@@ -1,23 +1,42 @@
 import { createMaterialTopTabNavigator } from '@react-navigation/material-top-tabs';
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import Routes from '@/navigation/routesNames';
 import { deviceUtils } from '@/utils';
-import { AddWalletSheet } from '@/screens/AddWalletSheet';
+import { AddWalletSheet, AddWalletSheetParams } from '@/screens/AddWalletSheet';
 import { ImportSeedPhraseSheet } from '@/screens/ImportSeedPhraseSheet';
 import { IS_ANDROID } from '@/env';
 import { SheetHandleFixedToTopHeight, SlackSheet } from '@/components/sheet';
 import { BackgroundProvider } from '@/design-system';
 import { StatusBar, View } from 'react-native';
+import { RouteProp, useRoute } from '@react-navigation/core';
+import { useNavigation } from '@/navigation';
 
 const Swipe = createMaterialTopTabNavigator();
 
-export const contentHeight =
-  deviceUtils.dimensions.height - SheetHandleFixedToTopHeight;
+type RouteParams = {
+  AddWalletNavigatorParams: AddWalletSheetParams;
+};
 
 export const AddWalletNavigator = () => {
+  const {
+    params: { isFirstWallet, userData },
+  } = useRoute<RouteProp<RouteParams, 'AddWalletNavigatorParams'>>();
+  const { setParams } = useNavigation();
+
+  const [sheetHeight, setSheetHeight] = useState(0);
+
+  useEffect(() => {
+    setParams({ sheetHeight, backgroundOpacity: sheetHeight ? undefined : 1 });
+  }, [setParams, sheetHeight]);
+
   const [scrollEnabled, setScrollEnabled] = useState(false);
 
+  const contentHeight =
```

Comment:
if a custom height is provided, that takes precedence, otherwise default to full screen sheet